### PR TITLE
karate: init at 1.2.0

### PR DESCRIPTION
**Pull request by kephas**
Co-authored-by: maxime.a.didier@gmail.com

###### Motivation for this change

Add a new tool

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kephas** [comment]
I'm not sure where to put the license or other metadata when creating the derivation with `pkgs.writeShellScriptBin`.

**Comment by sikmir** [comment]
Why not `mkDerivation` with `meta`?

**Comment by kephas** [comment]
I added metadata.

**Review by 06kellyjac — CHANGES_REQUESTED** [review]
Thanks for your contribution.

I'm not very familiar with packaging jar's but I found a package called tessera that pretty much does the same packaging method

There were a couple big changes I was suggesting and the output looked very hard to read so I've just got the whole change below (with some review explanation comments you should delete after):

(also please update this and your commit/pr message to init at `1.2.0` since there's been a new release)

```nix
# based on tessera package
{ lib, stdenvNoCC, fetchurl, jre, makeWrapper }:

# stdenvNoCC because you're not compiling anything
stdenvNoCC.mkDerivation rec {
  pname = "karate";
  version = "1.1.0";

  src = fetchurl {
    url = "https://github.com/karatelabs/karate/releases/download/v${version}/karate-${version}.jar";
    sha256 = "0pcm9dlps8s6k8pypm1ghhq3yjvivq60d8s2nk8hx0yrx2chdsa0";
  };
  dontUnpack = true;

  nativeBuildInputs = [ makeWrapper ];

  # just use within-build tools like makeWrapper
  # also since we're overwriting the installPhase we need to run the pre and post hooks
  installPhase = ''
    runHook preInstall
    makeWrapper ${jre}/bin/java $out/bin/karate --add-flags "-jar $src"
    runHook postInstall
  '';

  meta = with lib; {
    description = "API Test Automation Made Simple";
    # align longDescription like other packages
    longDescription = ''
      Karate is the only open-source tool to combine API test-automation, mocks,
      performance-testing and even UI automation into a single, unified
      framework. The BDD syntax popularized by Cucumber is language-neutral, and
      easy for even non-programmers. Assertions and HTML reports are built-in,
      and you can run tests in parallel for speed.
    '';
    homepage = "https://github.com/karatelabs/karate";
    changelog = "https://github.com/karatelabs/karate/releases/tag/v${version}";
    license = licenses.mit;
    maintainers = [ maintainers.kephasp ];
    # set platform based on the main dependency
    platforms = jre.meta.platforms;
  };
}
```

Output

```
λ ./result/bin/karate --help
07:57:33.456 [main]  INFO  com.intuit.karate - Karate version: 1.1.0
Usage: <main class> [-CDhsSW] [-d[=<debugPort>]] [-c=<cert>] [-e=<env>]
                    [-g=<configDir>] [-i=<importFile>] [-j=<jobServerUrl>]
                    [-k=<key>] [-n=<name>] [-o=<output>] [-p=<port>]
                    [-P=<prefix>] [-T=<threads>] [-w=<workingDir>] [-f=<formats>
                    [,<formats>...]]... [-H=<hookFactoryClassNames>[,
                    <hookFactoryClassNames>...]]... [-m=<mocks>[,
                    <mocks>...]]... [-t=<tags>]... [<paths>[$<paths>...]...]
      [<paths>[$<paths>...]...]
                            one or more tests (features) or search-paths to run
  -c, --cert=<cert>         ssl certificate (default: cert.pem)
  -C, --clean               clean output directory
  -d, --debug[=<debugPort>] debug mode (optional port else dynamically chosen)
  -D, --dryrun              dry run, generate html reports only
  -e, --env=<env>           value of 'karate.env'
  -f, --format=<formats>[,<formats>...]
                            comma separate report output formats. tilde
                              excludes the output report. html report is
                              included by default unless it's negated.e.g. '-f
                              json,cucumber:json,junit:xml. Possible values
                              [html: Karate HTML, json: Karate JSON, cucumber:
                              json: Cucumber JSON, junit:xml: JUnit XML]
  -g, --configdir=<configDir>
                            directory where 'karate-config.js' is expected
                              (default 'classpath:' or <workingdir>)
  -h, --help                display this help message
  -H, --hook=<hookFactoryClassNames>[,<hookFactoryClassNames>...]
                            class name of a RuntimeHook (or RuntimeHookFactory)
                              to add
  -i, --import=<importFile> import and convert a file
  -j, --jobserver=<jobServerUrl>
                            job server url
  -k, --key=<key>           ssl private key (default: key.pem)
  -m, --mock, --mocks=<mocks>[,<mocks>...]
                            one or more mock server files
  -n, --name=<name>         scenario name
  -o, --output=<output>     directory where logs and reports are output
                              (default 'target')
  -p, --port=<port>         server port (default 8080)
  -P, --prefix=<prefix>     mock server path prefix (context-path)
  -s, --ssl                 use ssl / https, will use 'cert.pem' and 'key.pem'
                              if they exist in the working directory, or
                              generate them
  -S, --serve               app server using --workdir (experimental)
  -t, --tags=<tags>         cucumber tags - e.g. '@smoke,~@skipme' [@ignore is
                              always skipped by default]
  -T, --threads=<threads>   number of threads when running tests
  -w, --workdir=<workingDir>
                            working directory, defaults to '.'
  -W, --watch               watch (and hot-reload) mock server file for changes
```

**Comment by kephas** [comment]
@06kellyjac  All your suggestions should be applied.

File: pkgs/development/tools/karate/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib, stdenvNoCC, pkgs, fetchurl, jre, makeWrapper }:
```
Comment (06kellyjac) [inline pkgs/development/tools/karate/default.nix:1]:
```suggestion
{ lib, stdenvNoCC, fetchurl, jre, makeWrapper }:
```

File: pkgs/development/tools/karate/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+stdenvNoCC.mkDerivation rec {
+  pname = "karate";
+  version = "1.2.0";
```
Comment (06kellyjac) [inline pkgs/development/tools/karate/default.nix:5]: Please update the PR title to match

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -6570,6 +6570,12 @@
     githubId = 2029444;
     name = "James Kent";
   };
+  kephasp = {
+    email = "pierre@nothos.net";
+    github = "kephas";
+    githubId = 762421;
+    name = "Pierre Thierry";
+  };
```
Comment (06kellyjac) [inline maintainers/maintainer-list.nix:6578]: This needs to be in its own commit. See contributing

**Review by 06kellyjac — CHANGES_REQUESTED** [review]
Small changes required

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (kephas) [reply maintainers/maintainer-list.nix:6609]: I didn't find the relevant part in `CONTRIBUTING.md` but I imitated previous commits on that file for the commit message.

**Comment by kephas** [comment]
@06kellyjac  Thanks for the review.

**Review by 06kellyjac — APPROVED** [review]
